rest-api/etl/etl.sql

```diff
@@ -1187,49 +1226,137 @@ TRUNCATE TABLE cdm.care_site;
+    site.site_id                            AS care_site_source_value,
     NULL                                    AS place_of_service_source_value,
     'care_site'                             AS unit_id
-FROM cdm.src_meas src_meas
-JOIN rdr.site site
-    ON  site.site_id = src_meas.finalized_site_id
+FROM rdr.site site
 ;
 
 -- -------------------------------------------------------------------
 -- source_file: src/visit_occurrence.sql
 -- -------------------------------------------------------------------
 
 -- -------------------------------------------------------------------
--- table: cdm.visit_occurrence
+-- table: cdm.tmp_visits_src.
+-- 'physical_measurements_id' is an id of a set of measurements during
+-- one patient's observation, so we use it as 'visit_occurence_id'.
+-- Min and Max measurements time over one physical_measurements_id
+-- is used as start and end visit dates.
 -- -------------------------------------------------------------------
-TRUNCATE TABLE cdm.visit_occurrence;
+DROP TABLE IF EXISTS cdm.tmp_visits_src;
 
-INSERT INTO cdm.visit_occurrence
+CREATE TABLE cdm.tmp_visits_src
+(
+    visit_occurrence_id bigint NOT NULL,
+    person_id bigint NOT NULL,
+    visit_start_datetime datetime NOT NULL,
+    visit_end_datetime datetime NOT NULL,
+    care_site_id bigint,
+    PRIMARY KEY (visit_occurrence_id)
+);
+
+INSERT INTO cdm.tmp_visits_src
 SELECT
     src_meas.physical_measurements_id       AS visit_occurrence_id,
     src_meas.participant_id                 AS person_id,
-    9202                                    AS visit_concept_id, -- 9202 - 'Outpatient Visit'
-    DATE(MIN(src_meas.measurement_time))    AS visit_start_date,
     MIN(src_meas.measurement_time)          AS visit_start_datetime,
-    DATE(MAX(src_meas.measurement_time))    AS visit_end_date,
     MAX(src_meas.measurement_time)          AS visit_end_datetime,
-    44818519                                AS visit_type_concept_id, -- 44818519 - 'Clinical Study Visit'
-    NULL                                    AS provider_id,
-    src_meas.finalized_site_id              AS care_site_id,
-    src_meas.physical_measurements_id       AS visit_source_value,
-    0                                       AS visit_source_concept_id,
-    'vis.survey'                            AS unit_id
+    src_meas.finalized_site_id              AS care_site_id
 FROM cdm.src_meas src_meas
 GROUP BY
+    src_meas.physical_measurements_id,
     src_meas.participant_id,
-    src_meas.finalized_site_id,
-    src_meas.physical_measurements_id
+    src_meas.finalized_site_id
+;
+
+-- -------------------------------------------------------------------
+-- table: cdm.tmp_visits_num
+-- -------------------------------------------------------------------
+DROP TABLE IF EXISTS cdm.tmp_visits_num;
+
+-- -------------------------------------------------------------------
+-- Here we order visits by person and visit_end_time, numbering each
```

Comment:
I don't understand why we need to do this. Can you elaborate?

Reply:
In cdm v.5.2 in visit_occurrence table there is a new field preceding_visit_occurrence_id. This part of the code calculates the preceding visit. This field is not required, so if you don’t need this field, we can leave it empty.
https://github.com/OHDSI/CommonDataModel/wiki/VISIT_OCCURRENCE

Reply:
I don't think we need this field. @jmandel let me know if you disagree. We can probably drop whatever logic in the ETL is computing it.